packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,410 @@
+    function scheduleTransaction(
+        Safe _safe,
+        uint256 _nonce,
+        ExecTransactionParams memory _params,
+        bytes memory _signatures
+    )
+        external
+    {
+        // Check that this guard is enabled on the calling Safe
+        if (!_isGuardEnabled(_safe)) {
+            revert TimelockGuard_GuardNotEnabled();
+        }
+
+        // Check that the guard has been configured for the Safe
+        if (!safeConfigs[_safe].configured) {
+            revert TimelockGuard_GuardNotConfigured();
+        }
+
+        // Get the encoded transaction data as defined in the Safe
+        // The format of the string returned is: "0x1901{domainSeparator}{safeTxHash}"
+        bytes memory txHashData = _safe.encodeTransactionData(
+            _params.to,
+            _params.value,
+            _params.data,
+            _params.operation,
+            _params.safeTxGas,
+            _params.baseGas,
+            _params.gasPrice,
+            _params.gasToken,
+            _params.refundReceiver,
+            _nonce
+        );
+
+        // Get the transaction hash and data as defined in the Safe
+        // This value is identical to keccak256(txHashData), but we prefer to use the Safe's own
+        // internal logic as it is more future-proof in case future versions of the Safe change
+        // the transaction hash derivation.
+        bytes32 txHash = _safe.getTransactionHash(
+            _params.to,
+            _params.value,
+            _params.data,
+            _params.operation,
+            _params.safeTxGas,
+            _params.baseGas,
+            _params.gasPrice,
+            _params.gasToken,
+            _params.refundReceiver,
+            _nonce
+        );
+
+        // Check if the transaction exists
+        // A transaction can only be scheduled once, regardless of whether it has been cancelled or not.
+        if (scheduledTransactions[_safe][txHash].executionTime != 0) {
+            revert TimelockGuard_TransactionAlreadyScheduled();
+        }
+
+        // Verify signatures using the Safe's signature checking logic
+        // This function call reverts if the signatures are invalid.
+        _safe.checkSignatures(txHash, txHashData, _signatures);
+
+        // Calculate the execution time
+        uint256 executionTime = block.timestamp + safeConfigs[_safe].timelockDelay;
+
+        // Schedule the transaction
+        scheduledTransactions[_safe][txHash] =
+            ScheduledTransaction({ executionTime: executionTime, cancelled: false, executed: false });
+
+        emit TransactionScheduled(_safe, txHash, executionTime);
+    }
+
+    /// @notice Returns the list of all scheduled but not cancelled transactions for a given safe
+    /// @dev MUST NOT revert - NOT IMPLEMENTED YET
+    /// @return List of pending transaction hashes
+    function checkPendingTransactions(address) external pure returns (bytes32[] memory) {
```

Comment:
```suggestion
    function getAllScheduledTransactions(address) external pure returns (bytes32[] memory) {
```

Or we can have `getScheduledTransaction` and `getScheduledTransactions`

Reply:
OK, I will assume that the full interface for those two functions is: 
```
getScheduledTransaction(Safe _safe, bytes32 _txHash) returns (ScheduledTransaction);
getScheduledTransactions(Safe _safe) returns (ScheduledTransaction[]);
```